license-upgrade/upgrade-license-plugin.groovy

```diff
@@ -287,66 +289,83 @@ boolean all = true
 if (type == Product.OPERATIONS_CENTER) {
     int plugins = 0
     int offline = 0
+    int tries = 20
+    long waitingFor = 1000
+    if (slowConnection) {
+        waitingFor = 3000
+    }
+
     jenkins.model.Jenkins.instance.getAllItems(com.cloudbees.opscenter.server.model.ConnectedMaster.class).each { master ->
         println "Analyzing " + master.name + "... "
         if(master.channel != null) {
-            def _masterStatus = parseMasterStatus(executeScript(script_status,master))
-            if (debug) {
-                print " " + _masterStatus
-            }
-            // If plugin requires update
-            if (_masterStatus[0] == '1') {
-                //print "Trying to upgrade plugin on " + master.name + "... "
-                _summary.append(master.name)
-                _summary.append(" - ")
-                boolean upgraded = performPluginUpdate(_masterStatus, direct, master)
-                if(upgraded) {
-                    if (debug) {
-                        print " Plugin upgraded successfully. "
-                    }
-                    _summary.append(" Plugin upgraded successfully, ")
-                    if (restart) { 
+            def _masterResult = executeScriptRemotely(master, script_status, tries, waitingFor)
+            if (_masterResult != null) {
+               
+                def _masterStatus = parseMasterStatus(_masterResult)
+                if (debug) {
+                    print " " + _masterStatus
+                }
+                if (_masterStatus.size() != 6) {
```

Comment:
isn't the status array pre-initialized to 6 elements, and if so, this check is not going to work?

Reply:
I know its late in the game, but we probably could have used an associative array (key, value) and maybe avoided this?